Content.Goobstation.Server/Damage/PreventDamageSystem.cs

```diff
@@ -0,0 +1,39 @@
+using Content.Shared.Damage;
+using Content.Shared.Mobs;
+using Content.Shared.Mobs.Systems;
+
+namespace Content.Goobstation.Server.Damage;
+
+/// <summary>
+/// This stops a mob from taking damage if that damge leads to its death
+/// </summary>
+public sealed class PreventDamageSystem : EntitySystem
+{
+    /// <inheritdoc/>
+    [Dependency] private readonly MobThresholdSystem _mobThreshold= null!;
```

Comment:
Do not trust rider and use `default!`
```suggestion
    [Dependency] private readonly MobThresholdSystem _mobThreshold= default!;
```